cmk/base/plugins/agent_based/cisco_asa_failover.py

```diff
@@ -0,0 +1,181 @@
+# URL   : https://thl-cmk.hopto.org
+# Date  : 2021-03-18
+#
+# rewrite of the original CMK2.0 check
+#
+#
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.4  'Failover LAN Interface'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.6  'Primary unit (this device)'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.2.7  'Secondary unit'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.4  2
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.6  9     < These two values flip during
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.3.7  10    < failover
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.4  'LAN_FO GigabitEthernet0/0.777'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.6  'Active unit'
+# .1.3.6.1.4.1.9.9.147.1.2.1.1.1.4.7  'Standby unit'
+
+# [['Failover LAN Interface', '2', 'LAN_FO GigabitEthernet0/0.777'],
+#  ['Primary unit', '9', 'Active unit'],
+#  ['Secondary unit (this device)', '10', 'Standby unit']]
+
+from typing import Dict, List
+
+from cmk.base.plugins.agent_based.agent_based_api.v1.type_defs import (
+    DiscoveryResult,
+    CheckResult,
+    StringTable,
+)
+
+from cmk.base.plugins.agent_based.agent_based_api.v1 import (
+    register,
+    Service,
+    Result,
+    State,
+    SNMPTree,
+    startswith,
+    contains,
+    any_of,
+)
+
+#
+# sample failover on
+#
+# [
+#  [
+#   ['Failover LAN Interface', '2', 'ClusterLink Port-channel4 (system)'],
+#   ['Primary unit (this device)', '9', 'Active unit'],
+#   ['Secondary unit', '10', 'Standby unit']
+#  ]
+# ]
+#
+# sample failover off --> _not configured_
+# [
+#  [
+#   ['Failover LAN Interface', '3', 'not Configured'],
+#   ['Primary unit', '3', 'Failover Off'],
+#   ['Secondary unit (this device)', '3', 'Failover Off']
+#  ]
+# ]
+#
+
+cisco_asa_state_names = {
+    '1': 'other',
+    '2': 'up',
+    '3': 'down',
+    '4': 'error',
+    '5': 'overTemp',
+    '6': 'busy',
+    '7': 'noMedia',
+    '8': 'backup',
+    '9': 'active',
+    '10': 'standby',
+}
+
+
+def get_cmk_status(st: int) -> State:
+    states = {
+        0: State.OK,
+        1: State.WARN,
+        2: State.CRIT
+    }
+    return states.get(st, State.CRIT)
+
+
+def parse_cisco_asa_failover(string_table: List[StringTable]) -> Dict:
+    def parse_line(line):
+        role = line[0].split(' ')[0].lower()
+        data = [role, line[1], line[2].lower()]
+        return data
+
+    parsed = {}
+    for line in string_table[0]:
+        if 'this device' in line[0].lower():
+            if line[2].lower() == 'failover off':
+                return {}
+            parsed['local'] = parse_line(line)
+        elif 'failover' in line[0].lower():
+            parsed['failover'] = line
+        else:
+            parsed['remote'] = parse_line(line)
+    return parsed
+
+
+#
+# {'failover': ['Failover LAN Interface', '2', 'ClusterLink Port-channel4 (system)'],
+#  'local': ['primary', '9', 'active unit'],
+#  'remote': ['secondary', '10', 'standby unit']}
+#
+def discovery_cisco_asa_failover(section: Dict) -> DiscoveryResult:
+    if not section == {}:
+        yield Service()
+
+
+def _convert_params(params):
```

Comment:
I decided that we completely drop this function, please simply remove it. I couldn't find this very old case it refers to in our code base, so I think we can safely ignore this.